### PR TITLE
Tokens, port polaris-react

### LINKED ISSUE #4088: Css variables
We use the "CSS only" approach to build on Polaris.
The documentation states including the css is sufficient (and using the classes in the HTML markup).
The current link provided is: https://unpkg.com/@shopify/polaris@6.2.0/dist/styles.css
However, this css file references a number of variables.
Could you please let us know where these variables are available so we can include them in our project?

### PR DESCRIPTION
**Pull request by romellogoodman**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris/issues/4088

This PR updates polaris-react to use `@shopify/polaris-tokens`. It removes the existing tokens from the polaris-react package, updates code to use the new token schema and removes tests and utilities that have already been ported over.

### CODE REVIEW

File: polaris-react/src/components/CustomProperties/CustomProperties.tsx
Comment (romellogoodman) [inline polaris-react/src/components/CustomProperties/CustomProperties.tsx:5]: Want to flag this as I'm sure if a direct input of the css file will have the effect we are looking for.

Comment (aaronccasanova) [reply polaris-react/src/components/CustomProperties/CustomProperties.tsx:5]: This is exact result I was hoping for! Definitely need a thorough tophat in Polaris and Web. Let me know if you need assistance with that.

Comment (romellogoodman) [reply polaris-react/src/components/CustomProperties/CustomProperties.tsx:5]: Fixed in this commit https://github.com/Shopify/polaris/pull/5659/commits/955da8e034fcc2b8e9e80077b56fcf0a153b3d40

File: .github/workflows/ci-stylelint-polaris.yml
Comment (aaronccasanova) [inline .github/workflows/ci-stylelint-polaris.yml]: Do we need to update this to `polaris-tokens/src/tokens/**` rather than deleting?

File: polaris-react/UNRELEASED.md
Comment (aaronccasanova) [inline polaris-react/UNRELEASED.md]:
```suggestion
- Integrated the `@shopify/polaris-tokens` CSS stylesheet in the `CustomProperties` component ([#5659](https://github.com/Shopify/polaris/pull/5659))
```

File: polaris-react/package.json
Comment (aaronccasanova) [inline polaris-react/package.json]: We should hold off merging this PR until we the stable `@shopify/polaris-tokens` library is released (e.g. `v5.0.0`).

File: polaris-react/src/components/CustomProperties/CustomProperties.tsx
Comment (aaronccasanova) [inline polaris-react/src/components/CustomProperties/CustomProperties.tsx:4]: This is a huge win! Nice work 🙌

File: polaris-react/src/components/CustomProperties/tests/CustomProperties.test.tsx
Comment (aaronccasanova) [inline polaris-react/src/components/CustomProperties/tests/CustomProperties.test.tsx:159]: It would be nice to have a test validating only one stylesheet is injected. I'm fine with this being a follow up task.

**Review by aaronccasanova — APPROVED** [review]
Amazing upgrade and clean up 👏 

The only blockers I see today are making sure we add the stable `v5.0.0` release of `@shopify/polaris-tokens` (as opposed to `v.5.0.0-rc.6`) and performing a thorough tophat in Web before merging.

File: .github/workflows/ci-stylelint-polaris.yml
Comment (romellogoodman) [reply .github/workflows/ci-stylelint-polaris.yml]: Restored. I thought they were related to the fact that stylelint used to directly use these files instead of the package

Comment (aaronccasanova) [reply .github/workflows/ci-stylelint-polaris.yml]: Oh right. I think that will apply if we adopt a monorepo tool that orchestrates interrelated version bumps. @samrose3 correct me if I'm wrong 🙏

Comment (sam-b-rose) [reply .github/workflows/ci-stylelint-polaris.yml]: Yep, good note here. Going to probably mash the CI together again soon as we have the proper tooling in place

File: polaris-react/src/components/CustomProperties/CustomProperties.scss
Comment (aaronccasanova) [inline polaris-react/src/components/CustomProperties/CustomProperties.scss]:
```suggestion
@import '../../../node_modules/@shopify/polaris-tokens/dist/css/styles.css
```

**Review by alex-page — APPROVED** [review]

File: polaris-react/UNRELEASED.md
Comment (aaronccasanova) [inline polaris-react/UNRELEASED.md]: Apologies, I didn't look what heading this changelog entry was under in my original suggestion.

Since there is quite a few changes in the PR I suggest we move this entry up to the `Enhancements` heading and add a "Upgraded `@shopify/polaris-tokens` to `v5.0.0`" entry for this heading `Dependency upgrades`.

Comment (aaronccasanova) [reply polaris-react/UNRELEASED.md]: Perhaps we should also include some deprecation notices for removing the `JSON` token-groups and the `src/tokens/tokens.ts` exports. They weren't documented as public APIs but definitely accessible:
- https://unpkg.com/browse/@shopify/polaris@9.0.0/build/esm/tokens/token-groups/
- https://unpkg.com/browse/@shopify/polaris@9.0.0/build/esm/tokens/tokens.js

**Review by yurm04 — APPROVED** [review]

**Review by kyledurand — APPROVED** [review]
👏

**Pull request merged**
